searx/user_help.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict
+import pkg_resources
+
+import flask
+from flask.helpers import url_for
+import mistletoe
+
+from . import get_setting
+from .version import GIT_URL
+
+HELP: Dict[str, str] = {}
+""" Maps a filename under help/ without the file extension to the rendered HTML. """
+
+
+def render(app: flask.Flask):
+    """
+    Renders the user documentation. Must be called after all Flask routes have been
+    registered, because the documentation might try to link to them with Flask's `url_for`.
+
+    We render the user documentation once on startup to improve performance.
+    """
+
+    link_targets = {
+        'brand.git_url': GIT_URL,
+        'brand.public_instances': get_setting('brand.public_instances'),
+        'brand.docs_url': get_setting('brand.docs_url'),
+    }
+
+    with app.test_request_context():
```

Comment:
To support in installed in a non root path:
`with app.test_request_context(base_url=get_setting('server.base_url')):`